x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -120,18 +121,33 @@ const getCspStatus = async ({
   };
 };
 
+export const statusQueryParamsSchema = schema.object({
+  /**
+   * CSP Plugin initialization includes creating indices/transforms/tasks.
+   * Prior to this initialization, the plugin is not ready to index findings.
+   */
+  check_initialized: schema.boolean({ defaultValue: false }),
+});
+
 export const defineGetCspStatusRoute = (router: CspRouter): void =>
   router.get(
     {
       path: STATUS_ROUTE_PATH,
-      validate: false,
+      validate: { query: statusQueryParamsSchema },
       options: {
         tags: ['access:cloud-security-posture-read'],
       },
     },
-    async (context, _, response) => {
+    async (context, request, response) => {
       const cspContext = await context.csp;
       try {
+        if (request.query.check_initialized) {
+          return response.ok({
+            body: {
+              initialized: cspContext.isPluginInitialized(),
+            },
+          });
+        }
```

Comment:
If you've decided not to use this and found a workaround (using the retry) lets remove the changes in the endpoint as they are redundant.

If you believe we should use the status endpoint for the testing and it is out of scope - lets open a ticket for it to fix that.

As for the changes, the naming is a bit not clear to me. Maybe it should be something like

`?check=init_only`
or `check=full`
and default can be `full`. So that the endpoint is either returning the full status or only parts of it like `initialize check only` or any other parts.

Reply:
changed to `check: "all" | "init"` query param